### ISSUE TITLE
Double text in parametertree action buttons on macOS (dark mode)

### ISSUE THREAD

**Issue by pbmanis** [pyqtgraph/pyqtgraph#2380]
### Short description
When creating a parameter tree in macOS dark mode, the text for the "action" buttons is doubled, with the two texts identical but offset. 

### Code to reproduce
MWE. Note that uncommenting the line "app.setStyle("Fusion") fixes the problem.

```python
import sys
import pyqtgraph as pg
from pyqtgraph.Qt import QtGui, QtCore, QtWidgets
import pyqtgraph.parametertree as pt

import numpy as np

app = pg.mkQApp()
# app.setStyle("Fusion")  # necessary to remove double labels on mac os
# note that the default:
# print("Current style: ", app.style().name())
# returns "macos".

class PGTest(QtWidgets.QWidget):
    def __init__(self):
        QtWidgets.QWidget.__init__(self)
        self.resize(320, 320)
        self.layout = QtWidgets.QGridLayout()
        self.layout.setContentsMargins(0, 0, 0, 0)
        self.setLayout(self.layout)
        self.ptree = pg.parametertree.ParameterTree(showHeader=False)
        self.layout.addWidget(self.ptree)
        self.params = pt.Parameter.create(name='params', type='group', children=[
            dict(name='Run/Stop', type='action', value=False),
        ])
        self.ptree.setParameters(self.params)
        self.params.sigTreeStateChanged.connect(self.params_changed)
        
        self.show()
    
    def params_changed(self, root, changes):
        # stub - nothing to do when testing
        pass

if __name__ == "__main__":
    win = PGTest()
    if sys.flags.interactive == 0:
        app.exec()
```

### Expected behavior
Buttons should have a single text item when the window style is "macOS".

### Real behavior
Buttons have two superimposed text items, slightly offset.

```
An error occurred?
There is no error raised.
```

### Tested environment(s)

 * PyQtGraph version: 0.12.4
 * Qt Python binding: PyQt6 6.3.1 Qt 6.3.1 (but also was present with PyQt5)
 * Python version: 3.9, 3.10
 * NumPy version: 1.21+
 * Operating system:  macOS 12.4
 * Installation method: pip, virtual environment

### Additional context
The double text does not appear when the window style is set to "Windows" or "Fusion".
The double text does not appear in pyqtgraph -m examples/parametertree.py (light window background).
The double text does not appear when running this under Windows with the same virtual environment requirements.

### Recommendation/Suggestion
The simple solution is to make the default style "Fusion" when detecting macOS. It is not clear that this behavior is introduced by pyqtgraph, but may reflect Qt/PyQt5/6 behavior.

**Comment by j9ac9k** [comment]
We've seen this (although I think this might be the first issue specific to this problem).

I think the text is doubled in all styles, but on macOS dark style there is some transparent layer making the doubled text visible.  I'm not sure what the fix actually is, and would very much welcome a PR to address it.

**Comment by pbmanis** [comment]
I agree, this looks like a transparent layer issue. In addition to using the "Fusion" style, I find the problem can be eliminated by setting app.setStyleSheet("QPushButton:!window {background: opaque;}"). Also, pyqtdarktheme (on pipy) app.setStyleSheet(qdarktheme.load_stylesheet()) works but adds a dependency.

I have been looking at the parameter tree code and the button text is written in a couple of  places, so wondering if that is related. I'm sorting through the logic there and will do a PR if I figure it out.

**Comment by j9ac9k** [comment]
Tagging @ntjess as he is our parameter tree expert among the active maintainers.

**Comment by ntjess** [comment]
The general problem is that an ActionParameterItem places its button over its `displayLabel`. An easy fix is to set the display label to blank text on this item, but I will investigate whether this is the correct general approach. In theory it should be fine, since there isn't a case where the button disappears leaving only the display label

**Comment by pbmanis** [comment]
@ntjess
You seem to have identified the issue! Changing the first line in ParameterItem's __init__ from:
        QtWidgets.QTreeWidgetItem.__init__(self, [param.title(), ''])
        to:
       QtWidgets.QTreeWidgetItem.__init__(self, ["", ''])
leaves the text behind the button cleared, and eliminates the double appearance.  The question is whether that data is referred to or used anywhere else, such as when doing a rename. I don't see anything obvious, nothing new broke in the unit/example tests, and my own usage works. However maybe @campagnola knows of the hidden traps.

**Comment by ntjess** [comment]
Great, but be warned -- the change should only occur for `ActionParameterItem` not the base `ParameterItem`, otherwise other types like string, int, etc. won't have their name show up.

**Comment by pbmanis** [comment]
Excellent point - in that case the correct place is in action.py, with a one-line insert. This works:

class ActionParameterItem(ParameterItem):
    """ParameterItem displaying a clickable button."""
    def __init__(self, param, depth):
        ParameterItem.__init__(self, param, depth)
-->   self.setText(0, "")   # clear the display label.
        self.layoutWidget = QtWidgets.QWidget()

**Comment by ntjess** [comment]
There's more to it than that, I think. You have to ensure that changes to `name` and `title` don't set the text back (i.e. `param.setOpts(name="A new name")` shouldn't recreate the text for that column. Otherwise, you're on the right track!

I will work on a PR when I have time that will address this, or you're welcome to create your own and I can review it.

**Comment by dkowollik** [comment]
Hi @ntjess,

did you find a solution for the issue above? In addition to this, I recognized that for long "action" button text, the text of column 0 is longer. Therefore, the last letters of the underlying text will show up. 
![grafik](https://github.com/user-attachments/assets/26c6aa2c-ac48-48aa-a6d2-2bfd8ae1e53b)
Depending on the letter width this gets worse. 
![grafik](https://github.com/user-attachments/assets/57de11c3-d60a-455a-95f9-990e3795674e)
It would be best to get rid of the underlying text.

**Comment by pgunn** [comment]
@pbmanis Is there a way to have applications reach into the underlying objects to set this without subclassing or changing pyqtgraph? I'm fine with iterating over each object I need to do this with, for now. I've tried a few different ways to reach into the underlying objects but I'm not getting it right. (I know the right solution is to have the issue fixed, just looking for a application-level workaround for now)

**Comment by pgunn** [comment]
@pbmanis Thanks, that setStyle call does the trick; I don't have a call to mkQApp() but I added:

`PySide6.QtWidgets.QApplication.setStyle("fusion")`

As the first thing in my app's main function before I make my QApplication and things look fine after that. Thanks; hopefully this workaround will be useful to people until the underlying issue is fixed.
